### DISCUSSION TITLE
How to craete logical partition with MBR?

### DISCUSSION THREAD

**Discussion by Yurii-huang** [storaged-project/blivet#1296] [Q&A] [answered]
How to craete logical partition with MBR?

**Comment by vojtechtrefny** [comment, marked as answer]
You can specify partition type with `part_type=parted.PARTITION_LOGICAL`.

Example of creating both the extended and logical partition:

```python
import blivet
import parted

b = blivet.Blivet()
b.reset()

sdf = b.devicetree.get_device_by_name("sdf")
b.format_device(sdf, blivet.formats.get_format("disklabel", label_type="msdos"))

epart = b.new_partition(parents=[sdf], size=blivet.size.Size("100 MiB"), part_type=parted.PARTITION_EXTENDED)
b.create_device(epart)

lpart = b.new_partition(parents=[sdf], size=blivet.size.Size("50 MiB"), part_type=parted.PARTITION_LOGICAL)
b.create_device(lpart)

blivet.partitioning.do_partitioning(b)

b.do_it()
```

**Reply by Yurii-huang** [reply]
thanks.

I tried to make a legacy boot disk, it seems that GPT can be used as a partition table, and a biosboot partition can be created. 
This is a feature of the GRUB2. 

Your comment may be helpful to others, this is a feature that is not mentioned on the API documentation.
